src/sentry/search/events/types.py

```diff
@@ -0,0 +1,13 @@
+from datetime import datetime
+from typing import List, Mapping, Union
+
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import BooleanCondition, Condition
+from snuba_sdk.function import CurriedFunction, Function
+
+WhereType = Union[Condition, BooleanCondition]
+# TODO: this should be a TypedDict instead
+ParamsType = Mapping[str, Union[List[int], int, str, datetime]]
+# Function is a subclass of CurriedFunction
+AggregateType = Union[CurriedFunction]
+SelectType = Union[Column, Function, CurriedFunction]
```

Comment:
Sigh unfortunately not:
```python
[E] Argument of type "List[SelectType] | None" cannot be assigned to parameter "select" of type "List[Column | CurriedFunction | Function] | None"
```

but AggregateType is probably redundant since it should just be `CurriedFunction` instead of `Union[CurriedFunction]` gonna continue using the alias for readability though with `SelectType`